### PR TITLE
US-7.5.1: Implement notifications portlet back office

### PR DESCRIPTION
**Pull request by SaraBoutej**

### CODE REVIEW

**Comment by azayati** [comment]
Related [commons PR](https://github.com/exodev/commons/pull/312)

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 1]
```diff
@@ -24,102 +24,128 @@
...
+        const dataId = $(`#${id}`).find('li').data('id');
+        this.notificationAction.id = dataId.toString();
+        this.notificationAction.operation = 'hide';
+        notificationlAPI.updateNotifications(this.notificationAction);
+      },
+      openDrawer() {
+        this.drawerNotification = !this.drawerNotification;
+        $('body').addClass('open-notif-drawer');
+        if(this.badge > 0) {
+          this.notificationOperation(null, 'resetNew');
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
          notificationlAPI.updateNotification(null, 'resetNew');
```

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 2]
```diff
@@ -24,102 +24,128 @@
...
+      notificationOperation(id, operation) {
+        this.notificationAction.id = id;
+        this.notificationAction.operation = operation;
+        notificationlAPI.updateNotifications(this.notificationAction);
+      },
+      hideNotification(id) {
+        const dataId = $(`#${id}`).find('li').data('id');
+        this.notificationAction.id = dataId.toString();
+        this.notificationAction.operation = 'hide';
+        notificationlAPI.updateNotifications(this.notificationAction);
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
        notificationlAPI.updateNotification(dataId.toString(), 'hide');
```
- Verify if .toString() is needed

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 3]
```diff
@@ -24,102 +24,128 @@
...
+            this.notificationAction.operation = 'markAsRead';
+            notificationlAPI.updateNotifications(this.notificationAction);
+            if(linkId.length >1 ) {
+              location.href = `${eXo.env.portal.context}/${eXo.env.portal.portalName}/${linkId[1]}`;
+            } else {
+              const id = linkId[0].split('/view_full_activity/')[1];
+              location.href = `${eXo.env.portal.context}/${eXo.env.portal.portalName}/activity?id=${id}`;
+            }
+      },
+      notificationOperation(id, operation) {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Remove this method, not needed

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 4]
```diff
@@ -24,102 +24,128 @@
...
+      this.getNotifListItem();
+      /*window.require(['SHARED/DWNotificationPopover'], function(dwNotificationPopover) {
+        dwNotificationPopover.init();*/
+      });
+    },
+    methods: {
+      getNotifListItem() {
+        notificationlAPI.getNotificationItems().then(
+                (data) => {
+                  this.items = data.notifications;
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
                  this.notifications= data.notifications;
```

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 5]
```diff
@@ -24,102 +24,128 @@
...
+    },
+    created() {
+      this.getNotifListItem();
+      /*window.require(['SHARED/DWNotificationPopover'], function(dwNotificationPopover) {
+        dwNotificationPopover.init();*/
+      });
+    },
+    methods: {
+      getNotifListItem() {
+        notificationlAPI.getNotificationItems().then(
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
        notificationlAPI.getNotifications().then(
```

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 6]
```diff
@@ -24,102 +24,128 @@
...
+      }
+    },
+    created() {
+      this.getNotifListItem();
+      /*window.require(['SHARED/DWNotificationPopover'], function(dwNotificationPopover) {
+        dwNotificationPopover.init();*/
+      });
+    },
+    methods: {
+      getNotifListItem() {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
      getNotifications() {
```

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 7]
```diff
@@ -24,102 +24,128 @@
...
+        right: null,
+        badge: 0,
+        notificationAction: {
+          id: "",
+          operation: ""
+        }
+      }
+    },
+    created() {
+      this.getNotifListItem();
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
      this.getNotifications();
```

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 8]
```diff
@@ -24,102 +24,128 @@
...
+      /*window.require(['SHARED/DWNotificationPopover'], function(dwNotificationPopover) {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Remove the commented lines

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 9]
```diff
@@ -24,102 +24,128 @@
...
-                </v-card>
-              </v-flex>
-            </v-layout>
-          </v-container>
+          <v-divider
+            :inset="inset" 
+            class="my-0"/>
+          <div class="notifDrawerItems">
+            <div
+              v-for="(item, i) in items"
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
              v-for="(notif, i) in notifications"
```

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 10]
```diff
@@ -24,102 +24,128 @@
...
+              :key="i"
+              :id="'notifItem-'+i"
+              class="notifDrawerItem">
+              <div
+                class="notifContent"
+                @click="getActivityId('notifItem-'+i)"
+                v-html="item.content_notif"></div>
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
                v-html="notif.content_notif"></div>
```

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 11]
```diff
@@ -24,102 +24,128 @@
...
+            <v-card 
+              flat
+              tile 
+              class="d-flex flex justify-end mx-2">
+              <v-btn 
+                text
+                small
+                class="text-uppercase caption "
+                color="primary"
+                @click="notificationOperation(null, 'markAllAsRead')">{{ this.$t('homepage.topBar.notification.MarkAllAsRead') }}</v-btn>
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
                @click="markAllAsRead()">{{ this.$t('homepage.topBar.notification.MarkAllAsRead') }}</v-btn>
```
- markAllAsRead method to be implemented

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 12]
```diff
@@ -24,102 +24,128 @@
...
+              <v-btn
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: This button should redirect to all notifications page

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 13]
```diff
@@ -24,102 +24,128 @@
...
-            background:"",
-
-          },
-        ],
+        items: {},
         mini: false,
-        right: null
+        right: null,
+        badge: 0,
+        notificationAction: {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Not needed, remove it

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 14]
```diff
@@ -24,102 +24,128 @@
...
-          },
-          {
-            avatar: 'https://cdn.vuetifyjs.com/images/lists/2.jpg',
-            title: "<span class='primary-color--text' span style='font-weight: bold'>Hamadi</span>  has commented on your task in <span style='font-weight: bold'>Maintenance Board</span> project",
-            subtitle: "1 day ago",
-            background:"",
-
-          },
-        ],
+        items: {},
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
        notifications: {},
```

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 15]
```diff
@@ -24,102 +24,128 @@
...
         mini: false,
-        right: null
+        right: null,
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Which need of right and mini variables ?

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 16]
```diff
@@ -24,102 +24,128 @@
...
+              v-for="(item, i) in items"
+              :key="i"
+              :id="'notifItem-'+i"
+              class="notifDrawerItem">
+              <div
+                class="notifContent"
+                @click="getActivityId('notifItem-'+i)"
+                v-html="item.content_notif"></div>
+                <span
+                :id="'notifClose-'+i"
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Which need of :id here ?

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 17]
```diff
@@ -24,102 +24,128 @@
...
+        notificationlAPI.getNotificationItems().then(
+                (data) => {
+                  this.items = data.notifications;
+                  this.badge = data.badge;
+                }
+        )
+      },
+      getActivityId(id) {
+            const dataLink = $(`#${id}`).find('.contentSmall:first').data('link');
+            const linkId = dataLink.split('/portal/intranet/');
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Verify if the activity link is always prefixed by /portal/intranet and not /portal/{currentSiteName}

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 18]
```diff
@@ -24,102 +24,128 @@
...
+            const dataId = $(`#${id}`).find('li').data('id');
+            this.notificationAction.id = dataId.toString();
+            this.notificationAction.operation = 'markAsRead';
+            notificationlAPI.updateNotifications(this.notificationAction);
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]:
```suggestion
            notificationlAPI.updateNotification(dataId.toString(), 'markAsRead');
```

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 19]
```diff
@@ -24,102 +24,128 @@
...
+                  this.items = data.notifications;
+                  this.badge = data.badge;
+                }
+        )
+      },
+      getActivityId(id) {
+            const dataLink = $(`#${id}`).find('.contentSmall:first').data('link');
+            const linkId = dataLink.split('/portal/intranet/');
+            const dataId = $(`#${id}`).find('li').data('id');
+            this.notificationAction.id = dataId.toString();
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Not needed, remove it

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 20]
```diff
@@ -24,102 +24,128 @@
...
+            this.notificationAction.operation = 'markAsRead';
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Not needed, remove it

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 21]
```diff
@@ -24,102 +24,128 @@
...
+            notificationlAPI.updateNotifications(this.notificationAction);
+            if(linkId.length >1 ) {
+              location.href = `${eXo.env.portal.context}/${eXo.env.portal.portalName}/${linkId[1]}`;
+            } else {
+              const id = linkId[0].split('/view_full_activity/')[1];
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: In which type of notification this case ?

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+        if(resp && resp.ok) {
+            return resp.json();
+        } else {
+            throw new Error(('Error when getting notification list'));
+        }
+    })
+}
+
+export function updateNotifications(notificationAction) {
+    return fetch(`/portal/rest/notifications/webNotifications/${notificationAction.operation}?user=${eXo.env.portal.userName}`, {
```
[the remark this answers is not in the document]
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+        headers: {
+            'Content-Type': 'text/plain'
+        },
+        credentials: 'include',
```
[the remark this answers is not in the document]
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js]: removed

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js [hunk 24]
```diff
@@ -0,0 +1,22 @@
...
+        method: 'PATCH',
+        body: notificationAction.id
```
[the remark this answers is not in the document]
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js]: Changed

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js [hunk 25]
```diff
@@ -0,0 +1,22 @@
+export function getNotificationItems() {
```
[the remark this answers is not in the document]
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+    return fetch(`/rest/notifications/webNotifications`, {
```
[the remark this answers is not in the document]
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js (hunk 25 above)
[the remark this answers is not in the document]
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/notificationsAPI.js]: ok

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 1 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: done

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 2 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: ok

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 3 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: changed by markAllAsRead method

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 4 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 5 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 7 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 8 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 9 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 10 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 11 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 12 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 13 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 14 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 15 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: not needed, i'll remove them!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 16 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: not needed!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 17 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: ok

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 18 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 19 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 20 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 21 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Kudos notification

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 27]
```diff
@@ -24,103 +24,179 @@
...
-            avatar: 'https://cdn.vuetifyjs.com/images/lists/2.jpg',
-            title: "<span class='primary-color--text' span style='font-weight: bold'>Hamadi</span>  has commented on your task in <span style='font-weight: bold'>Maintenance Board</span> project",
-            subtitle: "1 day ago",
-            background:"",
-
-          },
-        ],
-        mini: false,
-        right: null
+        notifications: {},
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Should be an array no ?

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 28]
```diff
@@ -24,103 +24,179 @@
...
+
+          const dataLink = $(this).find('.contentSmall:first').data('link');
+          const linkId = dataLink.split('/portal/intranet/');
+          const dataId = $(this).data('id').toString();
+
+          // ----------------- Mark as read
+
+          $(this).on('click', function(evt) {
+            evt.stopPropagation();
+            if($(this).hasClass('unread')) {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Try to refresh by calling this.getNotifications(); after calling notificationlAPI.updateNotification and reseting this.notifications and this.badge
-> to be tried for all actions

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 29]
```diff
@@ -24,103 +24,179 @@
...
+          // ------------- hide notif
+
+          $(this).find('.remove-item').off('click')
+                  .on('click', function(evt) {
+                    evt.stopPropagation();
+                    notificationlAPI.updateNotification(dataId,'hide');
+                    $(this).parents('li:first').slideUp(300);
+          });
+
+          // ------------- Accept connexion request
```
[the remark this answers is not in the document]
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue [hunk 30]
```diff
@@ -24,103 +24,179 @@
...
+                    restURl += `gtn:csrf=${  eXo.env.portal.csrfToken}`;
+                    if(restURl && restURl.length > 0) {
+                      $.ajax(restURl).done(function (data) {
+                        $(document).trigger("exo-invitation-updated");
+                      });
+                    }
+                    window.location.assign(`${eXo.env.portal.context}/${eXo.env.portal.portalName}/${linkId[1]}`);
+                  });
+
+          // ------------- Refuse connexion request
```
[the remark this answers is not in the document]
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 27 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: yes, changed!

File: digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue (hunk 28 above)
Comment (SaraBoutej) [reply digital-workplace-webapps/src/main/webapp/vue-app/topBar/notification/components/NotificationApp.vue]: done!

**Pull request merged**